code/modules/awaymissions/zlevel.dm

```diff
@@ -62,6 +62,13 @@ var/global/list/potentialRandomZlevels = generateMapList(filename = "config/away
 /proc/seedRuins(list/z_levels = null, budget = 0, whitelist = /area/space, list/potentialRuins = space_ruins_templates)
 	if(!z_levels || !z_levels.len)
 		z_levels = list(1)
+
+	for(var/zl in z_levels)
+		var/turf/T = locate(1, 1, zl)
+		if(!T)
+			WARNING("Z level [zl] does not exist - Not generating ruins")
+			return

```

Comment:
honestly it feels like this should be an all-or-nothing scenario, if you fail to include a z level that your code expects, that's a serious error and should be surfaced as such